src/box/sql/build.c

```diff
@@ -2848,8 +2848,10 @@ sql_create_index(struct Parse *parse) {
 	}
 	index->def->key_def->part_count = new_part_count;
 
-	if (!index_def_is_valid(index->def, def->name))
+	if (!index_def_is_valid(index->def, def->name)) {
```

Comment:
As a follow-up patch I would suggest to rename this function to `index_def_check_is_valid()`. Because this function has a side effect - it sets the diag.

Reply:
Thank you for the review! Done.